File: 백준/Silver/1929. 소수 구하기/소수 구하기.js
const fs = require("fs");

const reader = {
  data: fs
    .readFileSync(process.platform === "linux" ? 0 : "input.txt", "utf-8")
    .toString()
    .trim(),

  read() {
    return this.data;
  },

  readIntArray() {
    return this.read().split(" ").map(s => parseInt(s));
  }
};

function solve() {
  const [M, N] = reader.readIntArray();
  
  const sieve = new Array(N + 1).fill(true);
  sieve[0] = sieve[1] = false;

  for (let i = 2; i <= Math.sqrt(N); i++) {
    if (!sieve[i]) continue;
    for (let j = i + i; j <= N; j += i) {
      sieve[j] = false;
    }
  }

  const answer = [];

  for (let i = M; i <= N; i++) {
    if (sieve[i]) answer.push(i);
  }

  return answer.join("\n");
}

console.log(solve());

File: 백준/Silver/1929. 소수 구하기/소수 구하기.py
M, N = map(int, input().split())


def sieve_of_eratosthenes(n):  # 에라토스테네스의 체
    sieve = [True] * (n + 1)
    sieve[0] = False
    sieve[1] = False

    for i in range(2, int(n ** 0.5) + 1):
        if sieve[i] == True:
            for j in range(i+i, n + 1, i):
                sieve[j] = False

    return [i for i in range(2, n + 1) if sieve[i] == True]


prime_nums = sieve_of_eratosthenes(N)

for n in prime_nums:
    if n < M:
        continue
    print(n)
